### PR TITLE
[complex] `masked_fill`: Complex Autograd support and update masked_scatter skips.

### LINKED ISSUE #53523: Complex Autograd support for `torch.masked_fill` is currently untested
Complex Autograd for `torch.masked_fill` [is](https://github.com/pytorch/pytorch/blob/0ca029b22d17d236d34bcecad44b94b35b1af4bb/aten/src/ATen/native/cpu/IndexKernel.cpp#L218) [currently](https://github.com/pytorch/pytorch/blob/0ca029b22d17d236d34bcecad44b94b35b1af4bb/aten/src/ATen/native/TensorAdvancedIndexing.cpp#L960-L972) [supported](https://github.com/pytorch/pytorch/blob/0ca029b22d17d236d34bcecad44b94b35b1af4bb/tools/autograd/derivatives.yaml#L682-L689) but untested. 

Remove the existing [method_tests](https://github.com/pytorch/pytorch/blob/0ca029b22d17d236d34bcecad44b94b35b1af4bb/torch/testing/_internal/common_methods_invocations.py#L3563-L3570) (that aren't currently testing complex) and add an OpInfo test instead.

cc @ezyang @anjali411 @dylanbespalko @mruberry @aocsa @VitalyFedyunin @walterddr

### LINKED ISSUE #53608: Complex autograd is supported for torch.Tensor.masked_scatter_ but untested
These skips (which were added at the time since masked_fill still hadn't been ported on CUDA) should be removed now that the port is complete and we can add dispatch for complex types for masked_fill kernel on CUDA. https://github.com/pytorch/pytorch/blob/0a97712326682fa6e70d263e5e275a52c8dcaf38/torch/testing/_internal/common_methods_invocations.py#L2027
```
>>> dest = torch.randn(10, dtype=torch.cdouble, device='cuda', requires_grad=True).clone()
>>> src = torch.randn(10, dtype=torch.cdouble, device='cuda', requires_grad=True)
>>> mask = torch.tensor((0, 0, 0, 0, 1, 0, 1, 0, 1, 0), dtype=torch.bool, device='cuda')
>>> dest.masked_scatter_(mask, src)
tensor([ 0.4441-0.2455j,  0.2870-0.1043j, -0.6185+0.6146j, -0.6428+0.5322j,
         0.7472+0.1586j, -0.5039+2.1577j,  0.1907+0.7764j,  0.2066+0.6968j,
        -0.2815+0.2045j, -1.3827+1.7941j], device='cuda:0',
        dtype=torch.complex128, grad_fn=<MaskedScatterBackward>)
```

cc @ezyang @anjali411 @dylanbespalko @mruberry @aocsa

### PR DESCRIPTION
**Pull request by kshitij12345**
Reference Issue: #33152
Previous PR : https://github.com/pytorch/pytorch/pull/52035, https://github.com/pytorch/pytorch/pull/52483

Fixes : #53608
Fixes : #53523

**Note**: This PR is based on `ci-all/*` branch to ascertain that we don't break the master again.

### CODE REVIEW

**Comment by kshitij12345** [comment]
Still failing with misaligned error. 

https://app.circleci.com/pipelines/github/pytorch/pytorch/287233/workflows/facdc282-6a64-4a93-bdd1-650c205cc8b1/jobs/11649649

Adding skip with reference to https://github.com/pytorch/pytorch/issues/52437

**Comment by kshitij12345** [comment]
cc: @mszhanyi

**Review by anjali411 — APPROVED** [review]
I think it's ok to follow up on the windows misaligned address error in a follow-up PR

**Comment by kshitij12345** [comment]
As is, it will break master. The skip that i added is incorrect (have to fix that)

However, I am hopeful that https://github.com/pytorch/pytorch/pull/54246 will fix the issue. So i want to try rebasing after that one is in.

**Comment by kshitij12345** [comment]
@anjali411 
With https://github.com/pytorch/pytorch/pull/54596 in,
I think this is good to go.
Waiting for CI to pass.

Thanks!

**Review by anjali411 — APPROVED** [review]

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -1809,17 +1809,32 @@ def sample_inputs_floor_divide(op_info, device, dtype, requires_grad):
...
+        SampleInput(make_tensor((S, S), device, dtype, low=None, high=None, requires_grad=requires_grad),
                     args=(bernoulli_scalar().to(device),
-                          make_tensor((M, M), device, dtype, low=None, high=None, requires_grad=requires_grad))),
+                          make_tensor((S, S), device, dtype, low=None, high=None, requires_grad=requires_grad))),
+    )
+
+    return samples
+
+def sample_inputs_masked_fill(op_info, device, dtype, requires_grad):
+    def _make_tensor_helper(shape):
```
Comment (Lezcano) [inline torch/testing/_internal/common_methods_invocations.py:1828]: nit: Prefer `partial`. See e.g.
https://github.com/pytorch/pytorch/blob/c690ed0ae85982b00623a73512c0f947f9f8882b/torch/testing/_internal/common_methods_invocations.py#L862

I would also suggest to apply the same pattern in  the `masked_scatter` operation, as I believe it makes the code more readable.

Also, consider using a generator as in
https://github.com/pytorch/pytorch/blob/c690ed0ae85982b00623a73512c0f947f9f8882b/torch/testing/_internal/common_methods_invocations.py#L1709-L1723
Generators are not supported by `OpInfo` yet, but when they are, they will be the preferred way to write these functions.

Comment (kshitij12345) [reply torch/testing/_internal/common_methods_invocations.py:1828]: Noted. Thanks! Will keep that in mind :)

**Pull request closed without merging**